kf/bpfmap.go

```diff
@@ -60,39 +58,9 @@ func (b *BPFMap) Update(value string) error {
 	}
 	defer ebpfMap.Close()
 
-	// check values are single or multiple
-	s := strings.Split(value, ",")
-
-	if b.Type == ebpf.Hash {
-		// clear map elements
-		key := 0
-		val := 0
-		entries := ebpfMap.Iterate()
-		for entries.Next(unsafe.Pointer(&key), unsafe.Pointer(&val)) {
-			// Order of keys is non-deterministic due to randomized map seed
-			if err := ebpfMap.Delete(unsafe.Pointer(&key)); err != nil {
-				log.Warn().Err(err).Msgf("delete hash map for key %d failed", key)
-			}
-		}
-
-		for key, val := range s {
-			v, _ := strconv.ParseInt(val, 10, 64)
-			x := 1
-			log.Info().Msgf("updating map %s key %d mapid %d", b.Name, v, b.MapID)
-			if err := ebpfMap.Update(unsafe.Pointer(&v), unsafe.Pointer(&x), 0); err != nil {
-				return fmt.Errorf("update hash map element failed for key %d error %w", key, err)
-			}
-		}
-	} else if b.Type == ebpf.Array {
-		for key, val := range s {
-			v, _ := strconv.ParseInt(val, 10, 64)
-			log.Info().Msgf("updating map %s key %d mapid %d", b.Name, v, b.MapID)
-			if err := ebpfMap.Update(unsafe.Pointer(&key), unsafe.Pointer(&v), 0); err != nil {
-				return fmt.Errorf("update array map index %d %w", key, err)
-			}
-		}
-	} else {
-		return fmt.Errorf("unsupported map type")
+	log.Info().Msgf("updating map %s key %d mapid %d", b.Name, key, b.MapID)
```

Comment:
Map args kind of replacement of arguments to the program, its not addition. Please clean up the map before updating.